codec-http/src/main/java/io/netty/handler/codec/spdy/SpdyOrHttpChooser.java

```diff
@@ -1,177 +0,0 @@
-/*
- * Copyright 2014 The Netty Project
- *
- * The Netty Project licenses this file to you under the Apache License,
- * version 2.0 (the "License"); you may not use this file except in compliance
- * with the License. You may obtain a copy of the License at:
- *
- *   http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
- * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
- * License for the specific language governing permissions and limitations
- * under the License.
- */
-package io.netty.handler.codec.spdy;
-
-import io.netty.buffer.ByteBuf;
-import io.netty.channel.ChannelHandler;
-import io.netty.channel.ChannelHandlerContext;
-import io.netty.channel.ChannelInboundHandler;
-import io.netty.channel.ChannelPipeline;
-import io.netty.handler.codec.ByteToMessageDecoder;
-import io.netty.handler.codec.http.HttpObjectAggregator;
-import io.netty.handler.codec.http.HttpRequestDecoder;
-import io.netty.handler.codec.http.HttpResponseEncoder;
-import io.netty.handler.ssl.SslHandler;
-import io.netty.util.internal.StringUtil;
-
-import javax.net.ssl.SSLEngine;
-import java.util.List;
-
-/**
- * {@link ChannelInboundHandler} which is responsible to setup the {@link ChannelPipeline} either for
- * HTTP or SPDY. This offers an easy way for users to support both at the same time while not care to
- * much about the low-level details.
- */
-public abstract class SpdyOrHttpChooser extends ByteToMessageDecoder {
```

Comment:
I don't think we can remove in 4.1... we can only remove in 5.0 as we only allow adding new methods to interface in minor releases. 

For HTTP/2 removing sounds fine though